File: finstack-py/tests/golden/runners/pricing_cap_floor.py
"""Domain runner for `rates.cap_floor` golden fixtures."""

from __future__ import annotations

from tests.golden.schema import GoldenFixture

from .pricing_common import run_pricing_fixture


def run(fixture: GoldenFixture) -> dict[str, float]:
    """Run one cap/floor fixture through the Python bindings."""
    return run_pricing_fixture(fixture)

File: finstack/valuations/tests/golden/runners/pricing_cap_floor.rs
//! Domain runner for `rates.cap_floor` golden fixtures.

use crate::golden::runner::DomainRunner;
use crate::golden::schema::GoldenFixture;
use std::collections::BTreeMap;

/// Cap/floor golden runner.
pub struct CapFloorRunner;

impl DomainRunner for CapFloorRunner {
    fn run(&self, fixture: &GoldenFixture) -> Result<BTreeMap<String, f64>, String> {
        crate::golden::runners::pricing_common::run_pricing_fixture(fixture)
    }
}
